### DISCUSSION TITLE
Issues in getting started with eclair-cli

### DISCUSSION THREAD

**Discussion by mohitk108** [ACINQ/eclair#2699] [Developers]
I have bitcoin-25.0 installed with the following bitcoin.conf:
```
server=1
txindex=1
regtest=1

rpcuser=claddy
rpcpassword=H-QAlqka_JTPhCBIruTjV3mcXrFmazKsCPnT7dNWO-g

zmqpubrawblock=tcp://127.0.0.1:29000
zmqpubhashblock=tcp://127.0.0.1:29000
zmqpubrawtx=tcp://127.0.0.1:29001

fallbackfee=0.0002
```
Then I installed eclair node from [releases](https://github.com/ACINQ/eclair/releases/tag/v0.9.0)
My eclair.conf file is:
```
eclair.chain = "regtest"
eclair.node-alias=claddy
eclair.server.port=9735

eclair.trampoline-payments-enable=true

eclair.api.enabled=true
eclair.api.port=9000
eclair.api.password=password

eclair.bitcoind.rpcport=18443
eclair.bitcoind.rpcuser=claddy
eclair.bitcoind.rpcpassword=H-QAlqka_JTPhCBIruTjV3mcXrFmazKsCPnT7dNWO-g
eclair.bitcoind.zmqblock="tcp://127.0.0.1:29000"
eclair.bitcoind.zmqtx="tcp://127.0.0.1:29001"
```
When I start the eclair node, I'm not getting any response:
![image](https://github.com/ACINQ/eclair/assets/37688684/84331db0-1e6e-41e1-97ae-65d9d271d5d9)
In a new terminal, when i try to connect to an endurance node as stated in the [Usage.md](https://github.com/ACINQ/eclair/blob/master/docs/Usage.md)
I' getting the following error:
`claddy@claddy:~$ eclair-cli connect --uri=03933884aaf1d6b108397e5efe5c86bcf2d8ca8d2f700eda99db9214fc2712b134@endurance.acinq.co:9735
Enter host password for user 'eclair-cli':
curl: (7) Failed to connect to localhost port 8080 after 0 ms: Connection refused`

I'm entering the password as `password`, as entered in the eclair.conf file.
How, do I proceed forward?

**Comment by sstone** [comment]
In your `eclair.conf` file you specify `eclair.api.port=9000` so you must also tell `eclair-cli` to use port 9000:
`eclair-cli -a localhost:9000 -p password connect ...`

Also, you've configured `bitcoind` and `eclair` to run in `regtest` mode (a local test network),  so you will not be able to connect to `endurance` which runs on `testnet`.

**Comment by t-bast** [comment]
> I have bitcoin-25.0 installed

Why do you use bitcoind 25.0? Our `README` says that you should use bitcoind 23.2, and that's what I told you when we last spoke about it.

> When I start the eclair node, I'm not getting any response:

This is perfectly normal, your node is running and does not print logs to the console. But if you look for the `eclair.log` file, you'll see that your node was started correctly.

> when i try to connect to an endurance node

As sstone explained, you're running on regtest, which is your own local test network, so you shouldn't try to connect to any node outside of your machine, that doesn't make sense. You should follow the instructions in https://github.com/t-bast/lightning-cfg/ to run multiple nodes on your local machine and connect them together.

**Reply by mohitk108** [reply]
### Update
- Fixed my bitcoind version. Now, it's 23.2
- I was able to setup 2 eclair node Ids, alice and bob and was able to do following:
- connect both nodes
- setup a channel between them
- create invoice for a `Test Payment`
- pay the invoice
- below is the log of `getsentinfo` command which describes it.
```
claddy@claddy:~$ eclair-cli -a localhost:9000 -p password getsentinfo --id=fceaf66b-55cd-4bee-bfa9-ff31f303e2f7
[
  {
    "id": "fceaf66b-55cd-4bee-bfa9-ff31f303e2f7",
    "parentId": "fceaf66b-55cd-4bee-bfa9-ff31f303e2f7",
    "paymentHash": "6375a03aeef166a277b11e75c703be152a09d9a2423015c3335224056162c561",
    "paymentType": "Standard",
    "amount": 100000,
    "recipientAmount": 100000,
    "recipientNodeId": "03e319aa4ecc7a89fb8b3feb6efe9075864b91048bff5bef14efd55a69760ddf17",
    "createdAt": {
      "iso": "2023-06-23T23:11:04.422Z",
      "unix": 1687561864
    },
    "invoice": {
      "prefix": "lnbcrt",
      "timestamp": 1687561790,
      "nodeId": "03e319aa4ecc7a89fb8b3feb6efe9075864b91048bff5bef14efd55a69760ddf17",
      "serialized": "lnbcrt1u1pjfvf37pp5vd66qwhw79n2yaa3re6uwqa7z54qnkdzggcptsen2gjq2ctzc4ssdq523jhxapqwpshjmt9de6qsp57gg0gql0lcay80ny35d9sg30s59wu68adjfjf3gn59zhmyc6y50qmqz9gxqrrsscqp79q7sqqqqqqqqqqqqqqqqqqqsqqqqqysgqalkgarqsgwmjqngaces28s5du5kmet20rwxtyg7xvw3gep7gzpf9ln0pwqg8e2w3mh2cxtmc8e9jf2hww2829y7s7vqd30g8pst8cycpqqmcza",
      "description": "Test payment",
      "paymentHash": "6375a03aeef166a277b11e75c703be152a09d9a2423015c3335224056162c561",
      "paymentMetadata": "2a",
      "expiry": 3600,
      "minFinalCltvExpiry": 30,
      "amount": 100000,
      "features": {
        "activated": {
          "trampoline_payment_prototype": "optional",
          "payment_secret": "mandatory",
          "basic_mpp": "optional",
          "option_payment_metadata": "optional",
          "var_onion_optin": "mandatory"
        },
        "unknown": []
      },
      "routingInfo": []
    },
    "status": {
      "type": "failed",
      "failures": [
        {
          "failureType": "Local",
          "failureMessage": "cannot route to self",
          "failedRoute": []
        }
      ],
      "completedAt": {
        "iso": "2023-06-23T23:11:04.426Z",
        "unix": 1687561864
      }
    }
  }
]
```

**Reply by t-bast** [reply]
Perfect, that shows you should now have a working setup! You should now do more tests, play around with all the APIs available for eclair (see https://acinq.github.io/eclair). For example, the response you just pasted shows that you did *not* succeed at making a payment (`"failureMessage": "cannot route to self"`). Do you understand why you got that error?

**Reply by mohitk108** [reply]
This error *probably* is generated because I tried to pay the invoice from 'Alice' to 'Alice'(back to itself). It's not possible to route a payment back to node sending it.

The payment was successful, when I create an invoice between Alice and Bob, and Bob pays it to Alice.

**Reply by t-bast** [reply]
That's exactly it :+1:

**Comment by mohitk108** [comment]
Does [rbfopen](https://acinq.github.io/eclair/#open-2) work in regtest mode?
```bash
claddy@claddy:~$ eclair-cli -a localhost:9000 -p password rbfopen --channelId=b91b499c7274a91b8fe698ac57df59513c89ace694e2fa7e3661641caeeca070 --targetFeerateSatByte=50000
cannot execute command=rbf in state=WAIT_FOR_FUNDING_CONFIRMED
```
After I mined blocks using `bitcoin-cli -generate 6`
then I'm getting the following output
```bash
claddy@claddy:~$ eclair-cli -a localhost:9000 -p password rbfopen --channelId=b91b499c7274a91b8fe698ac57df59513c89ace694e2fa7e3661641caeeca070 --targetFeerateSatByte=50000
cannot execute command=rbf in state=NORMAL
```

**Reply by t-bast** [reply]
`rbfopen` only works for dual-funded channels, which is fully implemented by eclair but not yet activated because the specification hasn't been accepted yet (https://github.com/lightning/bolts/pull/851). You can think of dual-funded channels as the next version of the protocol to open channels between lightning nodes, but you can ignore it for now since it's not activated by default.

The goal of that API is to increase the fees of an _unconfirmed_ transaction. Once the transaction is confirmed (which is your second test, after `bitcoin-cli generate 6`), it wouldn't make sense to try to increase the fees.
